keys/signer_secp256r1.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package keys
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+
+	"github.com/coinbase/rosetta-sdk-go/asserter"
+	"github.com/coinbase/rosetta-sdk-go/types"
+)
+
+// SignerSecp256r1 is initialized from a keypair
+type SignerSecp256r1 struct {
+	KeyPair *KeyPair
+}
+
+// The Ecdsa signature is the couple (r, s), both r and s are 32 bytes
+const (
+	EcdsaRLen   = 32
+	EcdsaSLen   = 32
+	EcdsaMsgLen = 32
+)
+
+var _ Signer = (*SignerSecp256r1)(nil)
+
+// PublicKey returns the PublicKey of the signer
+func (s *SignerSecp256r1) PublicKey() *types.PublicKey {
+	return s.KeyPair.PublicKey
+}
+
+// Sign arbitrary payloads using a KeyPair
```

Comment:
nit: document signature format. per below, it seems to be `R || S`

Reply:
Updated